Left turn gap report

avetraffic opened the pull request.

Commit:
Create outline for Left Turn Gap Report

Commit:
Create outline for Left Turn Gap Report

Commit:
Add more data to each approachResult, add more properties to FinalGapAnalysisReportViewModel, clean up final report.

Commit:
Add more data to each approachResult, add more properties to FinalGapAnalysisReportViewModel, clean up final report.

Commit:
Add additional values, logic to control what fields are viewed

Commit:
Add additional values, logic to control what fields are viewed

Commit:
Add spinners to button

Commit:
Add spinners to button

Commit:
Add additional logic for SIgnal Type and CrossProductReview

Commit:
Add additional logic for SIgnal Type and CrossProductReview

Commit:
Update charts, add additional styling for PDF

Commit:
Update charts, add additional styling for PDF

Commit:
Change Bootstrap path from absolute to relative

Commit:
Change Bootstrap path from absolute to relative

Commit:
Add % symbol to PedVsFailures chart

Commit:
Add % symbol to PedVsFailures chart

Commit:
Change line colors, refactor stuff

Commit:
Change line colors, refactor stuff

Commit:
Fix position of tables

Commit:
Fix position of tables

Commit:
Separate AM and PM reports, move most logic from GetFinalGapAnalysisReport into GetApproachResult, general refactor

Commit:
Separate AM and PM reports, move most logic from GetFinalGapAnalysisReport into GetApproachResult, general refactor

Commit:
change true case to GapOutOk, PedCycleOk, and LeftTurnVolumeOk to success

Commit:
change true case to GapOutOk, PedCycleOk, and LeftTurnVolumeOk to success

Commit:
Update chart intervals

Commit:
Update chart intervals

Commit:
Add opposingApproach , peak period time, gap chart height

Commit:
Add opposingApproach , peak period time, gap chart height

Commit:
Add peak description and opposing approach, general refactor

Commit:
Add peak description and opposing approach, general refactor

Commit:
Change report bin size to 15

Commit:
Change report bin size to 15

Commit:
Add Get24HourPeriod to FinalGapAnalysisReportParameters, refactor left turn controller

Commit:
Add Get24HourPeriod to FinalGapAnalysisReportParameters, refactor left turn controller

Commit:
Fix mistake in signal data check report, css changes

Commit:
Fix mistake in signal data check report, css changes

Commit:
Add Left Turn Report to menu, add auth roles restrictions, rename peak period description

Commit:
Add Left Turn Report to menu, add auth roles restrictions, rename peak period description

Commit:
Add additional styling, remove opposingApproach function

Commit:
Add additional styling, remove opposingApproach function

Commit:
Add GetSignalType to Signal class

Commit:
Add GetSignalType to Signal class

Commit:
Update menu with authorize action link helper

Commit:
refactor

Commit:
Add approach and opposing approach descriptions

Commit:
merge stuff

Commit:
Refactor FinalGapAnalysisReport

Commit:
Add extension method for grabbing description from enum, add methods to grab phaseType and signalHeadType, use in LeftTurnGapController

Commit:
Add project too

dloweAvenueConsultants committed:
Merge branch 'master' into AddActionLinkFilterPackage

dloweAvenueConsultants reviewed (changes requested):
1. ATSPM.Application.Reports/Business/LeftTurnGapReport/LeftTurnGapOutAnalysis.cs  Line 58 You changed the minutes from 30 to 15 minutes. At one point they wanted this at 30 minutes. Let's ask them to make sure this is what they want.
2. MOE.Common/Migrations/Configuration.cs Line 592 The Left turn gap report has already been added to the menu. No need to add it again.
3. SPM/Models/LeftTurnGapReport/SplitFailResultViewModel.cs Splt fail does not have an opposing direction. It analyzes all approaches. Not sure how we want to handle this. It should only come in to play if it is the on analysis selected.

Commit:
Move stop spinner function to success/fail

Commit:
Update graphs from 30 -> 15 minutes

Commit:
remove unneeded import

Commit:
Remove duplicate left turn gap menu

Commit:
small fix

Commit:
Remove opposingDirection from split fail analysis

avetraffic commented:
1. I asked them and they decided on 15 minutes. I've updated all the other graphs to reflect that.
2. Good catch
3. Removed all my changes, and updated the report to show nothing if split fail is the only option selected

Commit:
Change chart axis name to "% Cycles w/ Peds"

Commit:
Fix opposing volume issues

nuzhat5 committed:
number of opposing lanes fix

nuzhat5 committed:
changed splitfails from int to double

dloweAvenueConsultants committed:
Merge pull request #9 from avenueconsultants/AddActionLinkFilterPackage

Add action link filter package

dloweAvenueConsultants committed:
Merge branch 'Development4.2.5'

dloweAvenueConsultants committed:
merge from master
